tests/test_numbered_collection.py

```diff
@@ -115,6 +115,12 @@ def test_append_renumber(self):
         self.assertEqual(cell.number, 4)
         self.assertEqual(len(cells), size + 2)
 
+    def test_append_renumber_problems(self):
+        prob1 = copy.deepcopy(self.simple_problem)
+        prob2 = copy.deepcopy(self.simple_problem)
+        prob2.materials.remove(prob2.materials[2])
+        prob2.materials.append_renumber(prob1.materials[1])
```

Comment:
Assert that this append actually happened, and that the material was renumbered.

Reply:
[a83092d](https://github.com/idaholab/MontePy/pull/467/commits/a83092dbcbd7711a054997723904c7d8fd4bc422)